app/shared/business_logic.py

```diff
@@ -1,26 +1,35 @@
-# TODO: temporary file for this code, maybe other code belongs here, maybe not. do decide
-
-
 import datetime
+from typing import Union
 
 from sqlalchemy.orm import Session
 
 from app.shared.db import worker_crud
 
 
-def get_store_archive_until(db: Session, group_id: str) -> datetime.datetime:
+# TODO: temporary file for this code, maybe other code belongs here, maybe not. do
+#  decide
+
+
+def get_store_archive_until(
+    db: Session, group_id: str
+) -> Union[datetime.datetime, None]:
```

Comment:
The typing was incorrect without out the `Union` due to the part that returns `None`.